### ISSUE TITLE
duplication + company-kind combinator

### ISSUE THREAD

**Issue by jdtsmith** [minad/cape#13]
Just noticed two issues when super-capf'ing dabbrev + elisp backend, ala:

```elisp
    (setq-local completion-at-point-functions
		`(,(cape-super-capf
		    #'cape-dabbrev
		    (cape-capf-with-predicate
		     #'elisp-completion-at-point
		     #'my/ignore-elisp-keywords))
		  cape-file)
```

1. Results from a single CAPF are doubling-up.  I don't know if this is specific to the dabbrev pairing, but it does not appear to be dabbrev and normal CAPF each having the same candidate (though it could be).
1. If one capf supports company-kind, and the other doesn't, kind-icon cannot tell the difference between "I don't do kind" and "I do kind, but not for this candidate".  Right now it's putting up an ugly "unknown" icon, but even if I correct that to make it a "blank" icon, now I get blanks for everything, even if an entire candidate list has no kind.  I'm not sure of a good solution other than "don't combine CAPFs with disparate kind availability".  I know this one is solved in Emacs 28 for example.  

Screen shot:

<img width="238" alt="image" src="https://user-images.githubusercontent.com/93749/143656851-fdcd9740-ab3e-45ae-880f-7dde079e341d.png">

**Comment by minad** [comment]
> Results from a single CAPF are doubling-up. I don't know if this is specific to the dabbrev pairing, but it does not appear to be dabbrev and normal CAPF each having the same candidate (though it could be).

Oh, right. This can happen. I will fix this and drop duplicates, preferring the candidate from the first capf.

> If one capf supports company-kind, and the other doesn't, kind-icon cannot tell the difference between "I don't do kind" and "I do kind, but not for this candidate". Right now it's putting up an ugly "unknown" icon, but even if I correct that to make it a "blank" icon, now I get blanks for everything, even if an entire candidate list has no kind.

You should correct the ugly icon. You won't get blanks for everything. cape-super-capf uses the appropriate company-kind function for each candidate, except if you have duplicates, there it is unclear which one to use.

**Comment by jdtsmith** [comment]
> EDIT: Fixed in 4b42415

Works thanks.  I also discovered (I hope) that order is relevant in the super-capf; by placing dabbrev last on the super-list, I get them as backups.  Unless this is an accident of HT ordering?  If not, worth adding to the docstring.  

> You should correct the ugly icon. You won't get blanks for everything, only for the candidates which don't have an associated kind function. 

I now just show the kind=t unknown-ish icon for the missing ones.  So you get no kind icon at all only when there is not a capf.  Not ideal, but better than blank/nothing, and will encourage people to ask backends to provide kinds.

**Comment by minad** [comment]
> Works thanks. I also discovered (I hope) that order is relevant in the super-capf; by placing dabbrev last on the super-list, I get them as backups. Unless this is an accident of HT ordering? If not, worth adding to the docstring.

Yes, the candidates are sorted by sub capf.
